### ISSUE TITLE
Crash in iOS SDK

### ISSUE THREAD

**Issue by matheda** [apptentive/apptentive-ios#195]
The latest version of our iOS app is using version 4.0.2 of the Apptentive SDK.  We are starting to see a good number of crashes that appear to be originating from a JSON serialization error in the SDK (see below).  Please let me know if any additional information would be helpful. Thanks!

```
Thread 1

Exception name: EXC_SOFTWARE / UNCAUGHT_NS_EXCEPTION
1
CoreFoundation
__exceptionPreprocess + 1232864
2
libobjc.A.dylib
objc_exception_throw + 34100
3
CoreFoundation
+[NSException raise:format:] + 1232676
4
Foundation
+[NSJSONSerialization JSONObjectWithData:options:error:] + 492236
5
Apptentive
__39-[ApptentiveRequestOperation startTask]_block_invoke (ApptentiveRequestOperation.m:116)
6
CFNetwork
__75-[__NSURLSessionLocal taskForClass:request:uploadFile:bodyData:completion:]_block_invoke + 13128
7
CFNetwork
__49-[__NSCFLocalSessionTask _task_onqueue_didFinish]_block_invoke + 110660
8
Foundation
__NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 940048
9
Foundation
-[NSBlockOperation main] + 173932
10
Foundation
-[__NSOperationInternal _start:] + 109348
11
Foundation
__NSOQSchedule_f + 949164
12
libdispatch.dylib
_dispatch_client_callout + 6556
13
libdispatch.dylib
_dispatch_queue_serial_drain + 64208
14
libdispatch.dylib
_dispatch_queue_invoke + 21192
15
libdispatch.dylib
_dispatch_root_queue_drain + 72268
16
libdispatch.dylib
_dispatch_worker_thread3 + 71628
17
libsystem_pthread.dylib
_pthread_wqthread + 4348
18
libsystem_pthread.dylib
start_wqthread + 3240
```

**Comment by frankus** [comment]
Thanks for the detailed report! We're taking a look at this now. 

Is this an occasional thing you're seeing from a crash reporting solution, or is it reproducible? If it's practical it would be helpful to have some information about the requests that are failing to decode.

**Comment by matheda** [comment]
We are seeing the crash in Firebase Crash Reporting, I am not able to reproduce it myself unfortunately.  The crash has occurred 66 total times in the past couple of days, all for unique users.  Based on the additional logging in the crash (see attached), it looks like the last call to engage might have been a simple string value ("Trail_Details_View") with no custom data.  Its hard to tell for certain.
![screen shot 2017-08-09 at 2 00 50 pm](https://user-images.githubusercontent.com/1895008/29136781-45264a28-7d0c-11e7-96ce-24c958c04bf4.png)

**Comment by matheda** [comment]
Hi, I wanted to check on the status of this issue.  This is our #1 crash at the moment (600+ instances).  We will be publishing a new version of our app this week and I would love to include a fix for this if it is available.

I took a look at the ApptentiveRequestOperation class and it looks like the SDK is getting an un-parseable response back from the server.  I would like to suggest wrapping the JSON parsing in a @try block to catch the exception when an unexpected response is received.

**Comment by frankus** [comment]
Sorry for not updating this earlier. We released a fix for this as part of our 4.0.3 release last Friday: https://github.com/apptentive/apptentive-ios/releases/tag/v4.0.3
